spark/src/main/scala/com/gu/tableversions/spark/VersionContext.scala

```diff
@@ -48,31 +49,31 @@ object SparkSupport {
     * Keep default (public) scope. It was `private` before and failed at Runtime (yet compiled...).
     * Have not tried with other scopes.
     */
-  def versionedInsertDatasetIntoTable[T](
-      versionContext: VersionContext,
+  def versionedInsertDatasetIntoTable[T, F[_]](
+      versionContext: VersionContext[F],
       dataset: Dataset[T],
       table: TableDefinition,
       userId: UserId,
-      message: String): IO[(TableVersion, TableChanges)] = {
+      message: String)(implicit F: Effect[F]): F[(TableVersion, TableChanges)] = {
```

Comment:
I think this could be `Sync` here as you only need to import, not run, side effects

Reply:
Indeed!